langchain4j-azure-cosmos-nosql/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/nosql/AbstractAzureCosmosDBNoSqlEmbeddingStore.java

```diff
@@ -0,0 +1,580 @@
+
+                            // Block to retrieve the first page synchronously
+                            List<JsonNode> documents = Objects.requireNonNull(
+                                            queryFlux.byPage(1).blockFirst())
+                                    .getResults();
+
+                            if (documents == null || documents.isEmpty()) {
+                                throw new IllegalArgumentException("No document found for id: " + id);
+                            }
+
+                            JsonNode document = documents.get(0);
+                            JsonNode metadataNode = document.get("metadata");
+
+                            if (metadataNode == null || metadataNode.get(metadataKey) == null) {
+                                throw new IllegalArgumentException("Partition key '" + metadataKey
+                                        + "' not found in metadata for document with id: " + id);
+                            }
+
+                            partitionKeyValue = metadataNode.get(metadataKey).asText();
+                        } else {
+                            throw new IllegalArgumentException(
+                                    "Unsupported partition key path: " + this.partitionKeyPath);
+                        }
+
+                        return CosmosBulkOperations.getDeleteItemOperation(id, new PartitionKey(partitionKeyValue));
+                    })
+                    .collect(toList());
+            // Execute bulk delete operations synchronously by using blockLast() on the
+            // Flux
+            this.container
+                    .executeBulkOperations(Flux.fromIterable(itemOperations))
+                    .doOnNext(response -> logger.info(
+                            "Document deleted with status: {}",
+                            response.getResponse().getStatusCode()))
+                    .doOnError(error -> logger.error("Error deleting document: {}", error.getMessage()))
+                    .blockLast();
+        } catch (Exception e) {
+            logger.error("Exception while deleting documents: {}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public void deleteContainer() {
+        try {
+            this.container.delete().block();
+            System.out.println("Deleted container: " + containerName);
+        } catch (CosmosException e) {
+            if (e.getStatusCode() == 404) {
+                System.out.println("Container does not exist: " + containerName);
+            } else {
+                System.out.println("Failed to delete container: " + e.getMessage());
+                throw e;
+            }
+        }
+    }
+
+    @Override
+    public EmbeddingSearchResult<TextSegment> search(final EmbeddingSearchRequest request) {
+        // Ensure topK is within acceptable limits
+        if (request.maxResults() > 1000) {
+            throw new IllegalArgumentException("Top K must be 1000 or less.");
```

Comment:
```suggestion
            throw new IllegalArgumentException("maxResults must be 1000 or less.");
```
Please change it everywhere